native/src/components/TextButton.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { ReactElement } from 'react'
+import { StyleProp, ViewStyle } from 'react-native'
+import styled from 'styled-components/native'
+
+import { CloseIcon } from '../assets'
+import Text from './base/Text'
+
+const TileButton = styled.TouchableOpacity<{ active: boolean }>`
+  background-color: ${props => (props.active ? props.theme.colors.themeColor : props.theme.colors.backgroundColor)};
+  padding: 8px;
+  align-items: center;
+  width: 100px;
+  height: 80px;
+  border-radius: 18px;
+  elevation: 5;
+  shadow-color: ${props => props.theme.colors.textColor};
+  shadow-offset: 0px 1px;
+  shadow-opacity: 0.2;
+  shadow-radius: 1px;
+  justify-content: space-around;
+`
+
+const TileButtonText = styled(Text)`
+  font-size: 12px;
+  color: ${props => props.theme.colors.textSecondaryColor};
+  font-family: ${props => props.theme.fonts.native.decorativeFontRegular};
+`
+
+const OverlayButton = styled.TouchableOpacity`
+  flex: 1;
+  flex-direction: row;
+  height: 30px;
+  padding: 4px 8px;
+  align-items: center;
+  margin: 0 4px;
+  border-radius: 20px;
+  background-color: ${props => props.theme.colors.backgroundColor};
+`
+
+const OverlayButtonIconContainer = styled.View`
+  height: 16px;
+  width: 16px;
+`
+
+const OverlayButtonText = styled(Text)`
+  color: ${props => props.theme.colors.textSecondaryColor};
+  font-family: ${props => props.theme.fonts.native.decorativeFontBold};
+`
+
+const OverlayButtonSpacer = styled.View`
+  width: 4px;
+`
+
+const Button = styled.TouchableOpacity<{ primary: boolean; disabled: boolean }>`
+  padding: 8px;
+  border-radius: 8px;
+  background-color: ${props => {
+    const buttonColor = props.primary ? props.theme.colors.themeColor : props.theme.colors.backgroundColor
+    return props.disabled ? props.theme.colors.textDecorationColor : buttonColor
+  }};
+`
+
+const ButtonText = styled.Text`
+  color: ${props => props.theme.colors.textColor};
+  font-weight: 500;
+  font-size: 18px;
+  text-align: center;
+`
+
+type ButtonSpecificProps =
+  | {
+      type: 'primary' | 'clear'
+      disabled?: boolean
+    }
+  | {
+      type: 'overlay'
+      Icon: ReactElement
+      closeButton?: boolean
+    }
+  | {
+      type: 'tile'
+      Icon: ReactElement
+      active: boolean
+    }
+
+type TextButtonProps = {
+  text: string
+  onPress: () => Promise<void> | void
+  style?: StyleProp<ViewStyle>
+} & ButtonSpecificProps
+
+const TextButton = ({ text, onPress, style, ...props }: TextButtonProps): ReactElement => {
```

Comment:
As this component does not share any code, this should be splitted to three components:
1) TextButton (for primary and clear)
2) OverlayButton
3) TileButton

Then all of them can have one matching type, avoiding these "or"-types and the switch-case. 
Then we can easily use at every point in the code
`<OverlayButton ... `
instead of
`<TextButton
          type='overlay'`
and auto complete feature of the IDE can do its job with adding needed.

These components can go to the `components/base` folder

Reply:
See #964, that was exactly what was critized there and I agree with the reviewers there. Therefore I now created one component.

Reply:
Well you got two approvals for this PR. 

@f1sh1918 whats your opinion on my comment above? or also @LeandraH and @sarahsporck What do you think is the better way to do this?

Reply:
My first inclination is that a single button is more likely to be used than three buttons. At the same time, I guess it might not be clear that the tiles are text buttons, and if someone re-implements those, that would be very annoying. So really, I don't have a clear preference.

I also think (hope?) that this button will be an interim component until we have a design system, at which point it might be a clearer decision point.

Reply:
In general I prefer smaller and expendable components. Therefore, having one component manage multiple use cases makes me feel uneasy already. In my understanding there should be separate components for separate use cases. Not that this solution is not expandable, but I think it can bloat up very quickly. I'm also curious to hear @f1sh1918 oppinion on this.

Reply:
sorry for the late response. I agree with steffi. I think these variants should only differ in some styling differences to keep in simple an clear. I think like "primary/secondary/clear (text only)" and if needed pass a `iconStart` / `iconEnd` prop.
What i criticized on #964 should be solved with including the slide button into the `TextButton`
The other button types differ too much in their behavior and should be defined in separate components

Reply:
Thanks for the discussion! I like separate components more as well :)